terraform/transform_destroy.go

```diff
@@ -193,6 +191,14 @@ func (t *CreateBeforeDestroyTransformer) Transform(g *Graph) error {
 		// This ensures that.
 		for _, sourceRaw := range g.UpEdges(cn).List() {
 			source := sourceRaw.(dag.Vertex)
+
+			// If the graph has a "root" node (one added by a RootTransformer and not
+			// just a resource that happens to have no ancestors), we don't want to
+			// add any edges to it, because then it ceases to be a root.
```

Comment:
Ah this makes a lot of sense. We now do a root transformer potentially before the destroy transform which never happened before. I see how this happened.